pipeline/bump-submodule/task

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+set -ex
+
+mkdir -p $HOME/.ssh
+ssh-keyscan -H github.com >> $HOME/.ssh/known_hosts
+
+pushd submodule-repo
+  target_sha="$(git rev-parse HEAD)"
+  submodule_repo_dir=${PWD}
+popd
+
+pushd "release-repo/$SUBMODULE_PATH"
+  git fetch ${submodule_repo_dir}
+  git checkout "$target_sha"
+  submodule_name=$(basename `git rev-parse --show-toplevel`)
+popd
+
+pushd release-repo
+  if [[ "$(git status --porcelain)" = "" ]]; then
```

Comment:
I'm not familiar with this - is it better than 
```if git diff-index --quiet HEAD --; then
    echo "No changes to commit"
fi```?

Reply:
I think that `git status --porcelain` is more readable than the other. Seeing if git status is empty is pretty clear but having to understand the exit code of diff-index is a little less obvious